libs/extensions/src/HistogramsFiller.cpp

```diff
@@ -82,4 +97,19 @@ void HistogramsFiller::FillTriggerVariablesPerTriggerSet(const std::shared_ptr<E
     if (passesDilepton) FillTriggerVariables(event, ttbarCategory, triggerSetName + "_dilepton");
     if (passesHadron) FillTriggerVariables(event, ttbarCategory, triggerSetName + "_hadron");
   }
+}
+
+void HistogramsFiller::FillHistograms1D(const std::shared_ptr<Event> event) {
+  for(auto &[histName, variableLocation] : histVariables) {
+    if(variableLocation[0] == "Event") {
+      // Assuming uint nObject from Event for now
+      uint eventVariable = event->Get(variableLocation[1]);
+      histogramsHandler->histograms1D[histName]->Fill(eventVariable);
+    } else {
+      auto collection = event->GetCollection(variableLocation[0]);
+      for (int i=0; i < event->GetNObjectsInCollection(variableLocation[0]); i++){
+        histogramsHandler->histograms1D[histName]->Fill(event->GetCollection(variableLocation[0])->at(i)->Get(variableLocation[1]));
+      }
```

Comment:
I think this should be possible to simplify with something like:

```C++
for(auto element : *collection){
  histogramsHandler->histograms1D[histName]->Fill(element->Get(variableLocation[1]));
}
```

And I don't think GetNObjectsInCollection needs to be defined.

Reply:
Ah ok, maybe I missed the star for the loop when I tried it - I'll try again